libs/elastic-indexing/src/lib/indexing.service.ts

```diff
@@ -43,4 +46,48 @@ export class IndexingService {
     logger.info('Done with sync')
     return true
   }
+
+  async rankEvaluation(locale: string) {
+    const requests = await this.getRankQueries(locale)
+    const index = await this.getIndex(locale)
+
+    return this.elasticService.precisionEvaluation(index, requests, 10)
+  }
+
+  protected async getIndex(locale: string) {
+    if (this.index) {
+      return this.index
+    }
+
+    const indexes = await this.elasticService
+      .getIndexFromAliases(`island-${locale}`)
+      .then((response) => response.map((index) => index.index))
+    // If no index were found, then default to _all
+    const index = indexes[0] ?? '_all'
+    this.index = index
+    return index
+  }
+
+  protected async getRankQueries(locale: string) {
+    if (this.rankQueries) {
+      return this.rankQueries
+    }
+
+    const index = await this.getIndex(locale)
+    this.rankQueries = rankQueries.map((entry) => {
+      const query = this.elasticService.getSearchQuery({
```

Comment:
You don't have to export `getSearchQuery` if this query generation is moved to the content-search lib.
Also getting search query might be handled in the rankEvaluation query generation by importing the searchQuery there